src/js/stores/AppsStore.js

```diff
@@ -146,12 +146,29 @@ function processApp(app) {
   return app;
 }
 
-function processApps(apps) {
-  return apps.map(function (app) {
-    return processApp(app);
+function processEmptyGroup(group) {
+  return Object.assign(processApp({id: group.id}), {
+    isGroup: true,
+    status: null
   });
 }
 
+function processAppsAndGroups(group, apps = []) {
```

Comment:
I kinda have the feeling, that this process is not working the way it should. I  expected to get a flat list off all groups and apps but the result only includes  "empty"  groups (including no apps but groups) and apps. Due to this implementation we need to re-extract groups from app and group ids when ever we need a full list of all items, which feels wrong.

*I've striped everything but id's and group indicator in the following examples.* 

**Result** 

```
[{
	"id": "/alpha"
}, {
	"id": "/beta"
}, {
	"id": "/gamma"
}, {
	"id": "/bar/baz"
}, {
	"id": "/foo",
	"isGroup": true
}, {
	"id": "/hello",
	"isGroup": true
}, {
	"id": "/hello/world/alpha"
}, {
	"id": "/hello/world/beta"
}]
```


**Expected Result**

```
[{
	"id": "/",
	"isGroup": true
}, {
	"id": "/alpha"
}, {
	"id": "/beta"
}, {
	"id": "/gamma"
}, {
	"id": "/bar",
	"isGroup": true
}, {
	"id": "/bar/baz"
}, {
	"id": "/foo",
	"isGroup": true
}, {
	"id": "/hello",
	"isGroup": true
}, {
	"id": "/hello/world",
	"isGroup": true
}, {
	"id": "/hello/world/alpha"
}, {
	"id": "/hello/world/beta"
}]
```


**Groups Response (v2/groups)**

```
{
	"id": "/",
	"dependencies": [],
	"version": "2016-03-04T13:38:49.215Z",
	"apps": [{
		"id": "/alpha"
	}, {
		"id": "/beta"
	}, {
		"id": "/gamma"
	}],
	"groups": [{
		"id": "/bar",
		"apps": [{
			"id": "/bar/baz"
		}],
		"groups": []
	}, {
		"id": "/foo",
		"apps": [],
		"groups": []
	}, {
		"id": "/hello",
		"apps": [],
		"groups": [{
			"id": "/hello/world",
			"apps": [{
				"id": "/hello/world/alpha"
			}, {
				"id": "/hello/world/beta"
			}],
			"groups": []
		}]
	}]
}
```